crates/subspace-farmer/src/single_disk_plot.rs

```diff
@@ -1077,17 +1074,21 @@ impl SingleDiskPlot {
     }
 
     /// Run and wait for background threads to exit or return an error
-    pub async fn run(mut self) -> anyhow::Result<()> {
+    pub fn run(&mut self) -> impl Future<Output = anyhow::Result<()>> + Send + Sync + 'static {
         if let Some(start_sender) = self.start_sender.take() {
             // Do not care if anyone is listening on the other side
             let _ = start_sender.send(());
         }
+        let error_receiver = self.error_receiver.take();
```

Comment:
The API with `&mut self` means you can call it multiple times, but on second call error receiver will be gone.

Reply:
Yeah, right. I also kinda like consuming single disk plot, but it is just so handy to query it in sdk apart from the issue related to drop.